hist/hist/src/TH1.cxx

```diff
@@ -4531,38 +4531,42 @@ TVirtualHistPainter *TH1::GetPainter(Option_t *option)
 
 ////////////////////////////////////////////////////////////////////////////////
 /// Compute Quantiles for this histogram
-/// Quantile x_q of a probability distribution Function F is defined as
+/// Quantile x_p := Q(p) is defined as the value x_p such that the cumulative
+/// probability distribution Function F of variable X yields:
 ///
 /// ~~~ {.cpp}
-///      F(x_q) = q with 0 <= q <= 1.
+///      F(x_p) = Pr(X <= x_p) = p with 0 <= p <= 1.
+///      x_p = Q(p) = F_inv(p)
 /// ~~~
 ///
 /// For instance the median x_0.5 of a distribution is defined as that value
-/// of the random variable for which the distribution function equals 0.5:
+/// of the random variable X for which the distribution function equals 0.5:
 ///
 /// ~~~ {.cpp}
-///      F(x_0.5) = Probability(x < x_0.5) = 0.5
+///      F(x_0.5) = Probability(X < x_0.5) = 0.5
+///      x_0.5 = Q(0.5)
 /// ~~~
 ///
+/// \author Eddy Offermann
 /// code from Eddy Offermann, Renaissance
 ///
-/// \param[in] nprobSum maximum size of array q and size of array probSum (if given)
-/// \param[in] probSum array of positions where quantiles will be computed.
-///   - if probSum is null, probSum will be computed internally and will
+/// \param[in] n maximum size of array p and size of array xp (if given)
+/// \param[out] xp array to be filled with nq quantiles evaluated at (p)
+/// \param[in] p array of probabilities where quantiles should be evaluated.
+///   - if p is null, p will be computed internally and will
 ///     have a size = number of bins + 1 in h. it will correspond to the
 ///     quantiles calculated at the lowest edge of the histogram (quantile=0) and
```

Comment:
This is not strictly to the changes asked and provided: if `p==nullptr`, then what is `in` and what is `out` is reversed, right? `xp` becomes just a list of edges already available otherwise, and `p` gets the CDF values at edges. This part is still not clear. Maybe some next ROOT version should remove default value for `p` and provide another function to return CDF values at edges?

Reply:
Sure, I know. I mean that the function does 2 very different things depending on the context. It either calculates xp given p or it calculates p. So it is either F^-1(p) or F(x)

Reply:
Added new issue. Sorry for the confusion